### PR TITLE
Add reason to allowUrl

### LINKED ISSUE #171: Add because to allowUrl
Nice to have a comment in allowUrl too, symmetry to allowDependency.

### PR DESCRIPTION
**Pull request by hfhbd**
Fixes https://github.com/cashapp/licensee/issues/171

Or do you want a base interface:
```kotlin
interface AllowOptions {
  fun because(reason: String)
}
```

### CODE REVIEW

File: src/main/kotlin/app/cash/licensee/pluginExtension.kt [hunk 1]
```diff
@@ -87,15 +92,6 @@ interface LicenseeExtension {
     allowDependency(groupId, artifactId, version, {})
   }
 
-  /** @suppress */
-  @JvmSynthetic // For Groovy build scripts, hide from normal callers.
-  fun allowDependency(
-    groupId: String,
-    artifactId: String,
-    version: String,
-    options: Closure<AllowDependencyOptions>,
```
Comment (hfhbd) [inline src/main/kotlin/app/cash/licensee/pluginExtension.kt:96]: Gradle generates an closure overload by itself if you use Action

File: src/main/kotlin/app/cash/licensee/pluginExtension.kt [hunk 2]
```diff
@@ -42,11 +41,17 @@ interface LicenseeExtension {
...
+   *     because 'is Apache-2.0'
+   *   }
    * }
    * ```
    */
-  fun allowUrl(url: String)
+  fun allowUrl(url: String, options: Action<AllowUrlOptions> = Action { })
+
+  /** @suppress */
+  @JvmSynthetic // For Groovy build scripts, hide from normal callers.
```
Comment (JakeWharton) [inline src/main/kotlin/app/cash/licensee/pluginExtension.kt:53]: Does `@JvmOverloads` on the other function work?

**Review by JakeWharton — APPROVED** [review]

File: src/main/kotlin/app/cash/licensee/pluginExtension.kt (hunk 2 above)
Comment (JakeWharton) [reply src/main/kotlin/app/cash/licensee/pluginExtension.kt:53]: I assume not since I didn't do it for the others

Comment (hfhbd) [reply src/main/kotlin/app/cash/licensee/pluginExtension.kt:53]: Nope, it doesn't work on interfaces.

**Pull request merged**